src/FSharp.Data.GraphQL.Server/Execution.fs

```diff
@@ -69,15 +69,17 @@ type NameValueLookup(keyValues: KeyValuePair<string, obj> []) =
     let rec structEq (x: NameValueLookup) (y: NameValueLookup) =
         if Object.ReferenceEquals(x, y) then true
         elif Object.ReferenceEquals(y, null) then false
+        elif Object.ReferenceEquals(x, null) then false
         elif x.Count <> y.Count then false
         else
             x.Buffer
             |> Array.forall2 (fun (a: KeyValuePair<string, obj>) (b: KeyValuePair<string, obj>) ->
                 if a.Key <> b.Key then false
                 else 
                     match a.Value, b.Value with
-                    | :? NameValueLookup, :? NameValueLookup as o -> structEq (downcast fst o) (downcast snd o)
-                    | :? seq<obj>, :? seq<obj> -> Seq.forall2 (=) (a.Value :?> seq<obj>) (b.Value :?> seq<obj>)
+                    | (:? NameValueLookup as x), (:? NameValueLookup as y) -> structEq x y
+                    | (:? seq<obj> as x), (:? seq<obj> as y) -> 
+                        if Seq.length x <> Seq.length y then false else Seq.forall2 (=) x y
```

Comment:
NameValueLookup structural equality was wrong when this assertion was made, ignoring differences in sequence lengths. If the first sequence (x) was a part of the second (contains a set of the first items), the test considered it equal, even if there was missing items in the end.